src/transaction/confirm.rs

```diff
@@ -298,7 +287,7 @@ mod tests {
         transport.add_response(generate_tx_receipt(hash, 2));
 
         let confirm = wait_for_confirmation(&web3, hash, ConfirmParams::mined())
-            .immediate()
+            .wait()
```

Comment:
I see now, this is probably related to how `Stream`s are implemented in `web3` that make it always yield.